jwst/dark_current/dark_sub.py

```diff
@@ -178,23 +275,38 @@ def subtract_dark(input, dark):
 
     """
 
+    dark_ints = dark.data.shape[0] -1
+
     log.debug("subtract_dark: nints=%d, ngroups=%d, size=%d,%d",
               input.meta.exposure.nints, input.meta.exposure.ngroups,
               input.data.shape[-1], input.data.shape[-2])
 
+
     # Create output as a copy of the input science data model
     output = input.copy()
 
+    
     # combine the science and dark DQ arrays
-    output.pixeldq = np.bitwise_or(input.pixeldq, dark.dq)
+    detector = input.meta.instrument.detector
+    if(detector[:3] == 'MIR'):
+           output.pixeldq = np.bitwise_or(input.pixeldq, np.bitwise_or(dark.dq[0,0,:,:], dark.dq[1,0,:,:]))
+    else:
+           output.pixeldq = np.bitwise_or(input.pixeldq, dark.dq)
 
 
     # loop over all integrations and groups in input science data
     for i in range(input.data.shape[0]):
-        for j in range(input.data.shape[1]):
+        if(detector[:3] =='MIR'):
+
+            if(i <= dark_ints): dark_int = dark.data[i]
```

Comment:
This doesn't look quite right. Assuming the dark data contain 2 integrations, line 278 sets dark_ints to 1. On the first iteration of the loop in line 298 i will be zero, and on the 2nd iteration it will be 1. So the "<=" comparison in line 301 will be true for the first TWO iterations of the loop, but you only want it to be true on the first, so that unique dark data get used only for the first integration of the science data. So I think line 301 should be:

  if(i < dark_ints):

and line 302 should be:

  if(i >= dark_ints):

Actually you don't even need that IF structure. It could be a simple IF/ELSE, because the only unique case is the first integration:

  if (i < dark_ints):
      dark_int = dark.data[i]
  else:
      dark_int = dark.data[dark_ints]